### PR TITLE
QuickSort Implementation updated to ES6+ syntax in JS

### LINKED ISSUE #1568: Updating the Quick Sort implementation .
@raghav-dalmia I am a participant in GSSoC, and I would like to work on the implementation of a quick sort algorithm in JS. The current approach uses ES5 syntax and would like to update to ES6  and use a recursive definition as well for its implementation using a functional programming approach.

### PR DESCRIPTION
**Pull request by abhu-A-J**
Fixes #1568 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- The previous implementation had ES5 syntax, I updated it to ES6+ and used recursive definition
 

#### Languages Used:
- JavaScript

#### Files Added:
- Only updated the file rather than adding a new one.

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by abhu-A-J** [comment]
@raghav-dalmia  @asha15 can you review it. If any changes are required let me know about it. Thanks

**Comment by asha15** [comment]
@abhu-A-J Your code has a syntax error. please correct it

File: Quick_Sort/Quick_Sort.js [hunk 1]
```diff
@@ -1,27 +1,36 @@
-function Quick_Sort(array, low, high){
+//Description: Function to perform the Quick Sort algorithm on an array of numbers
+//Expected Output: returns the sorted array
```
Comment (asha15) [inline Quick_Sort/Quick_Sort.js]:
```suggestion
//Expected Output: returns the sorted array
```
move this to the beginning of the file

File: Quick_Sort/Quick_Sort.js [hunk 2]
```diff
@@ -1,27 +1,36 @@
...
-      var t = array[temp];
-      array[temp] = array[i];
-      array[i] = t;
-      temp++
+//Helper function used to swap values based on indices given
+const swap = (arr, a, b)=>{
+    let temp = arr[a];
+    arr[a] = arr[b];
+    arr[b] = temp;
+}
```
Comment (asha15) [inline Quick_Sort/Quick_Sort.js]: keep a line space after this function

File: Quick_Sort/Quick_Sort.js [hunk 3]
```diff
@@ -1,27 +1,36 @@
...
+//Helper function to get the Pivoting Element Index
+const pivotIndex=(arr, start = 0, end = arr.length - 1)=>{
```
Comment (asha15) [inline Quick_Sort/Quick_Sort.js]: keep space before and after the equation

File: Quick_Sort/Quick_Sort.js [hunk 4]
```diff
@@ -1,27 +1,36 @@
...
+    let swapIndex = start;
+    let pivot = arr[start];  //considering start element is pivot for first run
+    for (let i = start + 1; i <= end; i++) {
+      if (pivot > arr[i]) {
+          swapIndex += 1;
```
Comment (asha15) [inline Quick_Sort/Quick_Sort.js]: correct the indentation

File: Quick_Sort/Quick_Sort.js [hunk 5]
```diff
@@ -1,27 +1,36 @@
...
+        let pivot = pivotIndex(arr, left, right); //get the index for picot elemenet
+        quickSort(arr, left, pivot - 1);  //run quickSort on all elements on left side of pivot index
+        quickSort(arr, pivot + 1, right); //run quickSort on all elements on right side of pivot index
+    }
+    return arr;
 }
 
-array = [4, 3, 6, 9, 5, 1, 8, 7, 2];
-Quick_Sort(array,0,array.length);
-console.log(array)
```
Comment (asha15) [inline Quick_Sort/Quick_Sort.js]: keep a line space after this

File: Quick_Sort/Quick_Sort.js [hunk 6]
```diff
@@ -1,27 +1,37 @@
-function Quick_Sort(array, low, high){
+//Description: Function to perform the Quick Sort algorithm on an array of numbers
```
Comment (SakshiShreya) [inline Quick_Sort/Quick_Sort.js]: Please add a space after //

File: Quick_Sort/Quick_Sort.js [hunk 7]
```diff
@@ -1,27 +1,37 @@
...
+}
 
-  for(var i = low + 1; i < high; ++i ){
-    if( array[i] < array[low] ){
-      var t = array[temp];
-      array[temp] = array[i];
-      array[i] = t;
-      temp++
+//Helper function to get the Pivoting Element Index
+const pivotIndex=(arr, start = 0, end = arr.length - 1)=>{
```
Comment (SakshiShreya) [inline Quick_Sort/Quick_Sort.js]: There should be a space around =>.

File: Quick_Sort/Quick_Sort.js [hunk 8]
```diff
@@ -1,27 +1,37 @@
...
-  Quick_Sort(array, low, temp-1);
-  Quick_Sort(array, temp, high);
+    swap(arr, start, swapIndex);
+    return swapIndex;
+}
 
+//Function that uses recursive definition for quick sort implementation
+const quickSort=(arr, left = 0, right = arr.length - 1)=>{
+    if (left < right) {
+        let pivot = pivotIndex(arr, left, right); //get the index for picot elemenet
```
Comment (SakshiShreya) [inline Quick_Sort/Quick_Sort.js]: Break the lines. One line can't contain more than 80 characters including the indentaton.

File: Quick_Sort/Quick_Sort.js [hunk 9]
```diff
@@ -1,27 +1,37 @@
...
+        quickSort(arr, left, pivot - 1);  //run quickSort on all elements on left side of pivot index
+        quickSort(arr, pivot + 1, right); //run quickSort on all elements on right side of pivot index
+    }
+    return arr;
 }
 
-array = [4, 3, 6, 9, 5, 1, 8, 7, 2];
-Quick_Sort(array,0,array.length);
-console.log(array)
+//I/P and O/P Examples
```
Comment (SakshiShreya) [inline Quick_Sort/Quick_Sort.js]: Move the input and output to the end of the file.

Comment (abhu-A-J) [reply Quick_Sort/Quick_Sort.js]: @shreya the red is the code that chnaged, the I/P examples are already at the end of the file

File: Quick_Sort/Quick_Sort.js (hunk 8 above)
Comment (abhu-A-J) [reply Quick_Sort/Quick_Sort.js]: Done with this change.

File: Quick_Sort/Quick_Sort.js (hunk 7 above)
Comment (abhu-A-J) [reply Quick_Sort/Quick_Sort.js]: Done with this change

File: Quick_Sort/Quick_Sort.js (hunk 6 above)
Comment (abhu-A-J) [reply Quick_Sort/Quick_Sort.js]: Done with this change

**Comment by abhu-A-J** [comment]
@singh-shreya6 Ma'am can you review the changes??

**Comment by abhu-A-J** [comment]
@raghav-dalmia Sir, can you review this PR??

File: Quick_Sort/Quick_Sort.js [hunk 10]
```diff
@@ -1,27 +1,45 @@
...
+// Helper function to get the Pivoting Element Index
+const pivotIndex=(arr, start = 0, end = arr.length - 1) => {
+    let swapIndex = start;
+    //considering start element is pivot for first run
+    let pivot = arr[start];  
+    for (let i = start + 1; i <= end; i++) {
+        if (pivot > arr[i]) {
+            swapIndex += 1;
+            swap(arr, swapIndex, i);
+      }
```
Comment (raghav-dalmia) [inline Quick_Sort/Quick_Sort.js]: indentation issue

**Comment by abhu-A-J** [comment]
@raghav-dalmia  fixed the indentatiion issue as mentioned. Kindly check it again if possible

**Pull request merged**